pkg/nodediscovery/nodediscovery.go

```diff
@@ -525,6 +528,65 @@ func (n *NodeDiscovery) mutateNodeResource(nodeResource *ciliumv2.CiliumNode) er
 				nodeResource.Spec.Azure.InterfaceName = c.Azure.InterfaceName
 			}
 		}
+
+	case ipamOption.IPAMAlibabaCloud:
+		nodeResource.Spec.AlibabaCloud = alibabaCloudTypes.Spec{}
+
+		instanceID, err := alibabaCloudMetadata.GetInstanceID(context.TODO())
+		if err != nil {
+			log.WithError(err).Fatal("Unable to retrieve InstanceID of own ECS instance")
+		}
+
+		if instanceID == "" {
+			return errors.New("InstanceID of own ECS instance is empty")
```

Comment:
Since this is the same error msg as in AWS ENI, let's create a variable:

```suggestion
			// Defined somewhere in this file
			var errEmptyInstanceID = errors.New(...)
			return errors.New("InstanceID of own ECS instance is empty")
```

Reply:
not quite same , ECS and EC2

Reply:
Ah, I didn't notice, very subtle :)